syntax = "proto3";

package armonik.api.grpc.v1.worker;

import "objects.proto";

option csharp_namespace = "ArmoniK.Api.gRPC.V1.Worker";
option java_package = "fr.aneo.armonik.api.grpc.v1.worker";

message ProcessRequest {
  string communication_token = 1;
  string session_id = 2;
  string task_id = 3;
  TaskOptions task_options = 4;
  repeated string expected_output_keys = 5;
  string payload_id = 6;
  repeated string data_dependencies = 7;
  string data_folder = 8;
  Configuration configuration = 9;
}

message ProcessReply {
  Output output = 1;
}

message HealthCheckReply {
  enum ServingStatus {
    UNKNOWN = 0;
    SERVING = 1;
    NOT_SERVING = 2;
  }
  ServingStatus status = 1;
}
